packages/shared/lib/core/app/classes/platform.class.ts

```diff
@@ -2,8 +2,21 @@ import { get } from 'svelte/store'
 
 import { IPlatform } from '../interfaces'
 import { mobile } from '../stores'
+import { PlatformExtension } from '../types'
 
 const IS_MOBILE = get(mobile)
 
 // TODO: https://github.com/iotaledger/firefly/issues/5143
-export const Platform: IPlatform = window[IS_MOBILE ? '__CAPACITOR__' : '__ELECTRON__']
+export let Platform: IPlatform = window[IS_MOBILE ? '__CAPACITOR__' : '__ELECTRON__']
+
+export function extendPlatform(extensions: PlatformExtension[]): void {
```

Comment:
This looks really hacky, maybe we should just defined a `Router` interface that takes care of handling all the routing logic in shared? It really smells like we need to make a distinction between `Platform` provided through electronApi and other logic that can be passed from the `desktop/lib` folders.